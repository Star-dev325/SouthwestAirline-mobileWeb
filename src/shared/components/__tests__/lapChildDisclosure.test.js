jest.mock('src/shared/bootstrap/urls', () => ({
  lapChildFAQ: '/lapChildFAQ'
}));

import React from 'react';
import LapChildDisclosure from 'src/shared/components/lapChildDisclosure';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'test/unit/helpers/configureMockStore';

describe('LapChildDisclosure', () => {
  describe('render', () => {
    it('should render adult text', () => {
      const { container } = createComponent();

      expect(container).toMatchSnapshot();
    });

    it('should render lap child text', () => {
      const { container } = createComponent({ paxNumber: 1, type: 'lapChild' });

      expect(container).toMatchSnapshot();
    });
  });

  describe('frequent traveler selected', () => {
    it('should render adult text with text-above in className', () => {
      const { container } = createComponent({ paxNumber: 1, type: 'adult', frequentTravelerId: '124XRS' });

      expect(container).toMatchSnapshot();
    });
  });

  describe('frequent traveler not selected', () => {
    it('should render adult text without text-above in className', () => {
      const { container } = createComponent({ paxNumber: 1, type: 'adult' });

      expect(container).toMatchSnapshot();
    });
  });

  const providerStore = configureMockStore()({
    app: {
      airBooking: {
        passengerCountData: {
          lapChildCount: 1,
          adultCount: 1,
          valueUpdated: false
        }
      }
    }
  });

  const createComponent = ({ paxNumber = 0, type = '', frequentTravelerId = '', ...otherProps } = {}) =>
    render(
      <Provider store={providerStore}>
        <LapChildDisclosure paxNumber={paxNumber} type={type} frequentTravelerId={frequentTravelerId} {...otherProps} />
      </Provider>
    );
});
