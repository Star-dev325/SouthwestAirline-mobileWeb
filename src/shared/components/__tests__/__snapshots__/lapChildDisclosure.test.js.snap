// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LapChildDisclosure frequent traveler not selected should render adult text without text-above in className 1`] = `
<div>
  <div
    class="lap-child-personal-info"
  >
    <div
      class="lap-child-personal-info-form"
    >
      <p>
        AIR_BOOKING__PASSENGERS__TITLE 1
      </p>
      <p
        class="lap-child-personal-info-form--helper-text"
      >
        SHARED__LAP_CHILD__TITLE_EXCLUDING_LAP_CHILD
      </p>
    </div>
  </div>
</div>
`;

exports[`LapChildDisclosure frequent traveler selected should render adult text with text-above in className 1`] = `
<div>
  <div
    class="lap-child-personal-info--text-above"
  >
    <div
      class="lap-child-personal-info-form"
    >
      <p>
        AIR_BOOKING__PASSENGERS__TITLE 1
      </p>
      <p
        class="lap-child-personal-info-form--helper-text"
      >
        SHARED__LAP_CHILD__TITLE_EXCLUDING_LAP_CHILD
      </p>
    </div>
  </div>
</div>
`;

exports[`LapChildDisclosure render should render adult text 1`] = `
<div>
  <div
    class="lap-child-personal-info"
  >
    <div
      class="lap-child-personal-info-form"
    >
      <p>
        AIR_BOOKING__PASSENGERS__TITLE 0
      </p>
      <p
        class="lap-child-personal-info-form--helper-text"
      >
        SHARED__LAP_CHILD__TITLE_EXCLUDING_LAP_CHILD
      </p>
    </div>
  </div>
</div>
`;

exports[`LapChildDisclosure render should render lap child text 1`] = `
<div>
  <div
    class="lap-child-personal-info"
  >
    <div
      class="lap-child-personal-info-form"
    >
      <p>
        AIR_BOOKING__PASSENGERS__TITLE 1
      </p>
      <p
        class="lap-child-personal-info-form--helper-text"
      >
        SHARED__LAP_CHILD__TITLE_LAP_CHILD
      </p>
    </div>
    <p
      class="lap-child-personal-info--helper-text"
    >
      AIR_BOOKING__PASSENGERS__LAPCHILD_INFO
      <a
        href="/lapChildFAQ"
        target="_blank"
      >
        AIR_BOOKING__LEARN_MORE
      </a>
    </p>
    <div
      class="lap-child-personal-info-form"
    >
      AIR_BOOKING__PASSENGERS__LAPCHILD_PROOF_TITLE
    </div>
    <p
      class="lap-child-personal-info--helper-text"
    >
      AIR_BOOKING__PASSENGERS__LAPCHILD_PROOF_INFO
    </p>
  </div>
</div>
`;
